controller_interface/test/test_controller_with_options.cpp

```diff
@@ -26,7 +26,7 @@ class FriendControllerWithOptions : public controller_with_options::ControllerWi
 TEST(ControllerWithOption, init_with_overrides) {
   // mocks the declaration of overrides parameters in a yaml file
   int argc = 8;
-  char const * const argv[argc] = {"", "--ros-args", "-p", "parameter_list.parameter1:=1.", "-p",
+  char const * const argv[8] = {"", "--ros-args", "-p", "parameter_list.parameter1:=1.", "-p",
```

Comment:
How about using a constexpr to avoid duplicating the magic number:

```
constexpr int argc = 8;
char const * const argv[argc] = {"", "--ros-args", "-p", ... };
rclcpp::init(arrlen(argv), argv);
```

Or better yet, just infer the length from the initializer list:
```
template<class T, size_t N>
constexpr size_t arrlen(T (&)[N]) { return N; }

...

char const * const argv[] = {"", "--ros-args", "-p", ... };
rclcpp::init(arrlen(argv), argv);
```

Reply:
Very much like that solution. I was just being lazy :P